host/sgx/exception_sim.c

```diff
@@ -0,0 +1,226 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <openenclave/host.h>
+#include <openenclave/internal/registers.h>
+#include <openenclave/internal/sgx/td.h>
+#include <signal.h>
+#if defined(_WIN32)
+#else
+#include <ucontext.h>
+#endif
+#include "enclave.h"
+#include "exception.h"
+
+#if defined(_WIN32)
+#else
+bool is_simulate(oe_host_exception_context_t* context)
+{
+    uint64_t tcs_address = context->rbx;
+    oe_enclave_t* enclave = oe_query_enclave_instance((void*)tcs_address);
```

Comment:
Handle case where enclave could be NULL.

Reply:
For my understanding, When is enclave NULL?

Reply:
It could happen in case where the signal is raised outside the enclave, even before an ecall is made. We register global signal handlers that will get called, and it is better to not make an assumption that an enclave exists.

Reply:
Fixed, thanks!